python/pyarrow/array.pxi

```diff
@@ -3925,7 +3925,8 @@ cdef class StructArray(Array):
             tosort = self
         indices = _pc().sort_indices(
             tosort,
-            options=_pc().SortOptions(sort_keys=[("", order)], **kwargs)
+            options=_pc().SortOptions(
+                sort_keys=[(field.name, order) for field in self.type], **kwargs)
```

Comment:
We should probably define this `sort_keys` also in the `if`/`else` clause above, so we can keep the `sort_keys=[("", order)]` for the case `by` was specified. 
This doesn't cause any issue right now, but that feels safer if we would be stricter in the future in `sort_indices` for a flat array, that we would be sorting it by multiple keys.

Reply:
I think I see what you mean, thanks! Pushed https://github.com/apache/arrow/pull/41495/commits/52b9d2c8cf7e6c16adee9246e30d01e1a1715700.